backend/chainlit/server.py

```diff
@@ -221,8 +227,30 @@ async def completion(request: CompletionRequest):
     return response
 
 
+@app.post("/login")
+async def login(form_data: OAuth2PasswordRequestForm = Depends()):
+    if not config.code.password_auth_callback:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="No auth_callback defined"
+        )
+
+    if config.code.password_auth_callback and not config.code.password_auth_callback(
+        form_data.username, form_data.password
+    ):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect login or password",
+        )
+
+    access_token = create_jwt({"id": "yeeees"})
```

Comment:
we should define the data struture for a user. maybe {
id: (like email)
role: enum
tags: optional list of tags
authprovider: password or oauth
}

Reply:
haha oops that was a test id 😂 

- yes for role
- what is the context for the tag property?
- I think the auth provider could be something created from within the chainlit code rather than provided by the user